src/Exception/RequestException.php

```diff
@@ -0,0 +1,55 @@
+<?php
+
+namespace Http\Client\Exception;
+
+use Psr\Http\Message\RequestInterface;
+
+/**
+ * Thrown when a request is available
```

Comment:
"Base exception for when a request failed. This can be because of network problems or the status code of the response."